### PR TITLE
Analytic Kerr ISCO calculation for EM-bright code

### PR DESCRIPTION
**Pull request by MarcoCusinato**
Optimization and correct handling of corner cases.
Running the code for 10^5 values of dimensionless black hole spin and spin inclination 30 times and averaging the results, we obtained a speedup of 3 times (from 52s to 17s), which is the same reported on the attached plots (run for a single 10^5 points cycle on a different machine).
The high spin case (\chi=1) has also been fixed (see chi1.png).
chi_inc_isso_new
chi_inc_isso_old
chi_inc_isso_rel_err
chi_inc_isso_rel_err_hist
chi1

### CODE REVIEW

File: pycbc/neutron_stars/pg_isso_solver.py [hunk 1]
```diff
@@ -26,59 +26,31 @@
 import numpy as np
 from scipy.optimize import root_scalar
 
+def ISCO_solution(chi, incl):
+    r"""Analytic solution of the innermost
+    stable circular orbit (ISCO).
```
Comment (titodalcanton) [inline pycbc/neutron_stars/pg_isso_solver.py:31]:
```suggestion
    stable circular orbit (ISCO) for the Kerr metric.
```

File: pycbc/neutron_stars/pg_isso_solver.py [hunk 2]
```diff
@@ -326,27 +298,19 @@ def PG_ISSO_solver(chi, incl):
...
-        for g0, sc in zip(initial_guess, sgnchi)])
+    rISCO_limit = ISCO_solution(chi, incl)
     # If the inclination is 0 or pi, just output the ISCO radius
     equatorial = np.isclose(incl, 0) | np.isclose(incl, np.pi)
     if all(equatorial):
         return rISCO_limit
 
     # ISSO radius for an inclination of pi/2
-    initial_guess = [9 if c < 0 else 6 for c in chi]
+    initial_guess = [5.27451056440629 if c > 0.5 else 6 for c in chi]
```
Comment (pannarale) [inline pycbc/neutron_stars/pg_isso_solver.py:313]: Can you insert a comment on the origin of this number?

**Comment by titodalcanton** [comment]
Many thanks @MarcoCusinato. For reference, could you report the speedup with respect to the previous code, and could you link or paste a plot of the relative error between the old and new code as a function of the entire spin/inclination space? Basically the plots we already discussed offline a while ago.

File: pycbc/neutron_stars/pg_isso_solver.py (hunk 2 above)
Comment (pannarale) [reply pycbc/neutron_stars/pg_isso_solver.py:313]: And at line 320: `polar = np.isclose(incl, np.pi / 2)` --> `polar = np.isclose(incl, 0.5*np.pi)`

**Review by pannarale — CHANGES_REQUESTED** [review]

**Comment by MarcoCusinato** [comment]
Running the code for 10^5 values of dimensionless black hole spin and spin inclination 30 times and averaging the results, we obtained a speedup of 3 times (from 52s to 17s), which is the same reported on the attached plots (run for a single 10^5 points cycle on a different machine).
The high spin case (\chi=1) has also been fixed (see chi1.png).

chi_inc_isso_rel_err
chi_inc_isso_new
chi_inc_isso_old
chi_inc_isso_rel_err_hist
chi1

File: pycbc/neutron_stars/pg_isso_solver.py [hunk 3]
```diff
@@ -27,58 +27,31 @@
...
-def ISCO_eq(r, chi):
-    r"""Polynomial that enables the calculation of the Kerr innermost
-    stable circular orbit (ISCO) radius via its roots,
+def ISCO_solution(chi, incl):
+    r"""Analytic solution of the innermost
+    stable circular orbit (ISCO) for the Kerr metric.
 
-    .. math:: Z(r) = [r (r-6)]^2 - \chi^2 [2r (3r+14) - 9 \chi^2]\,.
+    ..See eq. (2.21) of
+    `Bardeen, J. M., Press, W. H., Teukolsky, S. A. (1972)
```
Comment (pannarale) [inline pycbc/neutron_stars/pg_isso_solver.py:35]: I think this backtick is causing a failure in the automated tests.

```Warning, treated as error:
[51084](https://github.com/gwastro/pycbc/actions/runs/3421735975/jobs/5698287634#step:6:51085)
/home/runner/work/pycbc/pycbc/.tox/py-docs/lib/python3.10/site-packages/pycbc/neutron_stars/pg_isso_solver.py:docstring of pycbc.neutron_stars.pg_isso_solver:1:Inline interpreted text or phrase reference start-string without end-string.```

Comment (MarcoCusinato) [reply pycbc/neutron_stars/pg_isso_solver.py]: Fixed, I think that the problem was that it could not find the closing backtick

**Review by pannarale — APPROVED** [review]

**Pull request merged**